vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -105,8 +107,19 @@ func (r *recommender) UpdateVPAs() {
 			if !had {
 				metrics_recommender.ObserveRecommendationLatency(vpa.Created)
 			}
+			delete(r.clusterState.EmptyVPAs, vpa.ID)
 		} else {
 			vpa.Conditions.Set(vpa_types.RecommendationProvided, false, "", "")
+			firstObserved, ok := r.clusterState.EmptyVPAs[vpa.ID]
+			if !ok {
+				r.clusterState.EmptyVPAs[vpa.ID] = time.Now()
+			} else {
+				if firstObserved.Add(RecommendationMissingMaxDuration).Before(time.Now()) {
```

Comment:
I'd change the clusterState.EmptyVPAs meaning slightly so keep track the time from the last warning (and reset it inside this if) so we'd emit a warning for a problematic VPA twice an hour, not every minute. WDYT?

Reply:
Done.